arrow/arrow.py

```diff
@@ -1390,84 +1497,101 @@ def for_json(self):
 
     # math
 
-    def __add__(self, other):
+    def __add__(self, other: Union[timedelta, relativedelta]) -> "Arrow":
 
         if isinstance(other, (timedelta, relativedelta)):
             return self.fromdatetime(self._datetime + other, self._datetime.tzinfo)
 
-        return NotImplemented
+        # FIXME: change to raise
+        return NotImplemented  # type: ignore
```

Comment:
why this method returns `NotImplemented` class? I think we should raise `NotImplemented`

Reply:
I agree, I think this should be a `raise`. This is a good example of some ancient code bloat that has carried on with Arrow over the years.

Reply:
Actually, it turns out that this should actually be `return NotImplemented`: https://stackoverflow.com/questions/878943/why-return-notimplemented-instead-of-raising-notimplementederror